src/main/java/com/exasol/projectkeeper/ProjectFilesFitter.java

```diff
@@ -0,0 +1,75 @@
+package com.exasol.projectkeeper;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.logging.Log;
+
+/**
+ * This class fits the projects file structure.
+ */
+public class ProjectFilesFitter extends AbstractProjectFilesValidator {
+    private final Log log;
+
+    /**
+     * Crate a new instance of {@link ProjectFilesFitter}.
+     * 
+     * @param log logger
+     */
+    public ProjectFilesFitter(final Log log) {
+        super();
+        this.log = log;
+    }
+
+    /**
+     * Fit the file structure of the project.
+     * 
+     * @param projectDir     project's base directory
+     * @param enabledModules list of enabled modules
+     * @throws MojoFailureException in case fitting failed
+     */
+    public void fitProjectStructure(final File projectDir, final List<String> enabledModules)
+            throws MojoFailureException {
+        final FitterTemplateVisitor fitterVisitor = new FitterTemplateVisitor(this.log);
+        super.run(fitterVisitor, projectDir, enabledModules);
+        if (fitterVisitor.hadErrors()) {
```

Comment:
Can you return a boolean from run?

Reply:
That would move a implementation detail of the specific visitor to the abstract class.
Ath the moment both visitors in deed hav this field. But in the future there could be different one's that don't.